File: dotnet/xmscommon/framing/AMQMethodBodyFactory.cs
using System;
using jpmorgan.mina.common;
using log4net;

namespace OpenAMQ.Framing
{
    public class AMQMethodBodyFactory : IBodyFactory
    {
        private static readonly AMQMethodBodyFactory _instance = new AMQMethodBodyFactory();

        public static AMQMethodBodyFactory GetInstance()
        {
            return _instance;
        }

        #region IBodyFactory Members

        /// <summary>
        /// Creates the body.
        /// </summary>
        /// <param name="inbuf">The ByteBuffer containing data from the network</param>
        /// <returns></returns>
        /// <exception>AMQFrameDecodingException</exception>
        public IBody CreateBody(ByteBuffer inbuf)
        {
            return MethodBodyDecoderRegistry.Get(inbuf.GetUnsignedShort(), inbuf.GetUnsignedShort());
        }

        #endregion
    }
}

File: clients/java/src/org/openamq/framing/AMQMethodBodyFactory.java
package org.openamq.framing;

import org.apache.log4j.Logger;
import org.apache.mina.common.ByteBuffer;

public class AMQMethodBodyFactory implements BodyFactory
{
    private static final Logger _log = Logger.getLogger(AMQMethodBodyFactory.class);
    
    private static final AMQMethodBodyFactory _instance = new AMQMethodBodyFactory();
    
    public static AMQMethodBodyFactory getInstance()
    {
        return _instance;
    }
    
    private AMQMethodBodyFactory()
    {
        _log.debug("Creating method body factory");
    }

    public AMQBody createBody(ByteBuffer in) throws AMQFrameDecodingException
    {
        return MethodBodyDecoderRegistry.get(in.getUnsignedShort(), in.getUnsignedShort());        
    }
}
